tables_app/src/main/java/org/opendatakit/tables/activities/MainActivity.java

```diff
@@ -295,6 +299,25 @@ public void swapScreens(ScreenType newScreenType) {
       newFragment = mgr.findFragmentByTag(newScreenType.name());
       if (newFragment == null) {
         newFragment = new TableManagerFragment();
+        Bundle bundle = new Bundle();
+        String defaultValue = Constants.TABLE_SORT_ORDER.SORT_ASC.getCorrespondingName();
+        String sortingOrder = defaultValue;
+        if(mPropSingleton.getProperty(CommonToolProperties.KEY_PREF_TABLES_SORT_BY_ORDER) != null
+                || (!mPropSingleton.getProperty(CommonToolProperties.KEY_PREF_TABLES_SORT_BY_ORDER).equals("")) )
+          sortingOrder = mPropSingleton.getProperty(CommonToolProperties.KEY_PREF_TABLES_SORT_BY_ORDER);
+
+        bundle.putSerializable(CommonToolProperties.KEY_PREF_TABLES_SORT_BY_ORDER,  Constants.TABLE_SORT_ORDER.getCorrespondingEnum(sortingOrder)  );
+        newFragment.setArguments(bundle);
+      }
+      else {
+        String defaultValue = Constants.TABLE_SORT_ORDER.SORT_ASC.getCorrespondingName();
+        String sortingOrder = defaultValue;
```

Comment:
The logic to retrieve `KEY_PREF_TABLES_SORT_BY_ORDER` is the same in both cases, you could move it to before the if statement. Also, the default value is already handled by PropertiesSingleton you do not need to handle it here.